pkg/util/test/util.go

```diff
@@ -576,6 +594,167 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
+	}
+
+	// Validate images are correct
+	opVersion, err := GetPxOperatorVersion()
+	if err != nil {
+		return fmt.Errorf("failed to get operator version, Err: %v", err)
+	}
+	k8sVersion, err := GetK8SVersion()
+	if err != nil {
+		return fmt.Errorf("failed to get k8s version, Err: %v", err)
+	}
+
+	// check if px-versions config-map exists
+	var pxImageList map[string]string
+	config, _ := coreops.Instance().GetConfigMap(pxVersionsConfigmapName, cluster.Namespace)
+	if config != nil {
+		logrus.Debugf("px-versions config-map found, fetching images from config-map")
+		pxImageList, err = GetImagesFromConfigMap(config)
+		if err != nil {
+			return fmt.Errorf("failed to get images from config-map: [%v]", err)
+		}
+	} else {
+		pxSpecURL := reconstructSpecURL(cluster)
+		pxImageList, err = GetImagesFromVersionURL(pxSpecURL, k8sVersion)
+		if err != nil {
+			return fmt.Errorf("failed to get images from version URL: [%v]", err)
+		}
+	}
+
+	pxVersion := GetPortworxVersion(cluster)
+
+	var nodeWiperImage string
+	if opVersion.GreaterThanOrEqual(opVer23_10_2) {
+		if pxVersion.GreaterThanOrEqual(pxVer3_1) {
+			pxImage := cluster.Spec.Image
+			pxVersionStr := strings.Split(pxImage, ":")[len(strings.Split(pxImage, ":"))-1]
+			nodeWiperImage = fmt.Sprintf("portworx/oci-monitor:%s", pxVersionStr)
+		} else {
+			nodeWiperImage = pxImageList["nodeWiper"]
+		}
+	} else if opVersion.LessThan(opVer23_10_2) {
+		if pxVersion.LessThan(pxVer3_1) {
+			nodeWiperImage = pxImageList["nodeWiper"]
+		} else {
+			nodeWiperImage = defaultNodeWiperImage
+		}
+	}
+	logrus.Debugf("Expected nodeWiper image: [%s]", nodeWiperImage)
+	err = validateContainerImageInsidePods(cluster, nodeWiperImage, "px-node-wiper", pods)
+	if err != nil {
+		return fmt.Errorf("failed to validate NodeWiper image: [%v]", err)
+	}
+	logrus.Debugf("NodeWiper images match!")
+
+	return nil
+}
+
+func GetImagesFromConfigMap(config *v1.ConfigMap) (map[string]string, error) {
+	pxImageList := map[string]string{}
+	versionData, ok := config.Data["versions"]
+	if !ok {
+		return pxImageList, fmt.Errorf("versions key not found in config-map")
+	}
+	for _, line := range strings.Split(versionData, "\n") {
+		splitLine := strings.SplitN(strings.TrimSpace(line), ":", 2)
+		if len(splitLine) != 2 {
+			continue
+		}
+		key, value := splitLine[0], splitLine[1]
+		pxImageList[key] = value
+	}
+
+	return pxImageList, nil
+}
+
+func reconstructSpecURL(cluster *corev1.StorageCluster) string {
+	var pxUrl string
+	pxImage := cluster.Spec.Image
+	pxVersionStr := strings.Split(pxImage, ":")[len(strings.Split(pxImage, ":"))-1]
+	_, err := version.NewSemver(pxVersionStr)
+	if err != nil {
+		logrus.WithError(err).Warnf("Invalid PX version [%s] extracted from image name", pxVersionStr)
+		for _, env := range cluster.Spec.Env {
+			if env.Name == PxImageEnvVarName {
+				pxImage = env.Value
+				pxUrl = fmt.Sprintf("https://edge-install.portworx.com/%s", pxImage)
```

Comment:
I think you meant `pxVersionStr` instead of `pxImage`

Reply:
Correct! Updated.